packages/storage/src/indexed-db/updater.ts

```diff
@@ -41,10 +41,19 @@ export class IbdUpdater {
     const tables = updates.all.map(u => u.table);
     const tx = this.db.transaction(tables, 'readwrite');
 
+    const batchOperations = [];
+
+    // Batch all the updates into promises
     for (const update of updates.all) {
-      await tx.objectStore(update.table).put(update.value, update.key);
-      this.notifySubscribers(update);
+      const putOperation = tx
+        .objectStore(update.table)
+        .put(update.value, update.key)
+        .then(() => {
+          this.notifySubscribers(update);
+        });
+      batchOperations.push(putOperation);
     }
+    await Promise.all(batchOperations);
 
     await tx.done;
```

Comment:
thought: while I think the Promise.all() change is a good one, this doesn't change the atomicity of this operation. The transaction is building up the operations, but not actually committing anything until the even loop is released. The `await tx.done` tells us when that is. Perhaps you have a different understanding of this though?